numba/compiler.py

```diff
@@ -1158,3 +1174,11 @@ def py_lowering_stage(targetctx, library, interp, flags):
         # Prepare for execution
         cfunc = targetctx.get_executable(library, fndesc, env)
         return _LowerResult(fndesc, call_helper, cfunc=cfunc, env=env)
+
+
+def _reload_parfors():
+    """Reloader for cached parfors
+    """
+    # Re-initialize the parallel backend when load from cache.
+    from numba.npyufunc.parallel import _launch_threads
+    _launch_threads()
```

Comment:
Interestingly, I think that because the backend launch will determine the threadpool instance via reading of an environment variable, it will be possible to cache the function with one threadpool in use and reload it with another. This isn't a problem, as in, nothing will break immediately. However, it does raise the question of whether envvars should also be cached?

Reply:
Shouldn't we let users to pick a different threadpool backend in a new session?  If the threadpool backend options are cached, the cached options can conflict with the envvars in a new session.

Reply:
I guess there's two concerns, one is the case you note, which assumes that the cache is being used to speed up time-to-live, whereas another concern may be that the cache is being used on a shared file system for distributed work to speed up distributed time-to-live, in which case the original comment stands. Either way, doesn't need fixing here!